### ISSUE TITLE
NEI refuses to load recipes with server utilities 2.0.11 installed.

### ISSUE THREAD

**Issue by Jerphf** [GTNewHorizons/GT-New-Horizons-Modpack#15379]
### Your GTNH Discord Username

jerphf

### Your Pack Version

2.5.1_java17-21

### Your Server

SP

### Java Version

Java 17

### Type of Server

Single Player

### Your Expectation

I tried to check a recipe in NEI and through the questbook.

### The Reality

I was instead met with an error saying to check logs, This shows up in the textchat. 
![error](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/157259858/365ce645-90c3-4e7b-a418-160978991330)

### Your Proposal

I'm not sure. The error says to check logs, so here's the latest log I have on hand
[fml-client-latest.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/13999803/fml-client-latest.log)

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by Jerphf** [comment]
I'm new to how github works and how to submit this kinda stuff, so tell me if anythings wrong

**Comment by Lyfts** [comment]
Hey. Just to be sure I understand correctly, are you saying that you can only reproduce this with serverutils installed?

**Comment by Jerphf** [comment]
Correct, it happens only if I add server utilities. Without it installed it works correctly. The instance was downloaded from the wiki, I haven’t added any other mods.

**Comment by Lyfts** [comment]
That's really odd, it looks like something is causing a CME. To my knowledge SU isn't touching anything that NEI cares about.

Was it all recipes or just a particular one?
Did you change anything in SU or NEI config?
Did you do anything special before checking the recipe?

<details>
  <summary>The error in question</summary>

```
[20:27:14] [Client thread/ERROR] [NotEnoughItems/]: Error while looking up crafting recipe
[20:27:14] [Client thread/ERROR] [NotEnoughItems/]: outputId: item
[20:27:14] [Client thread/ERROR] [NotEnoughItems/]: results: [1xgt.metaitem.01@11308]
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: java.util.concurrent.ExecutionException: java.util.ConcurrentModificationException: java.util.ConcurrentModificationException: java.util.ConcurrentModificationException
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinTask.reportExecutionException(ForkJoinTask.java:605)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:981)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.recipe.RecipeHandlerQuery.getHandlersWithRecipes(RecipeHandlerQuery.java:83)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.recipe.RecipeHandlerQuery.getRecipeHandlersParallel(RecipeHandlerQuery.java:57)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.recipe.RecipeHandlerQuery.runWithProfiling(RecipeHandlerQuery.java:39)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.recipe.GuiCraftingRecipe.getCraftingHandlers(GuiCraftingRecipe.java:76)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.recipe.GuiCraftingRecipe.createRecipeGui(GuiCraftingRecipe.java:44)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.recipe.GuiCraftingRecipe.openRecipeGui(GuiCraftingRecipe.java:31)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.recipe.GuiCraftingRecipe.openRecipeGui(GuiCraftingRecipe.java:26)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.PanelWidget.mouseUp(PanelWidget.java:336)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.BookmarkPanel.mouseUp(BookmarkPanel.java:1096)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.LayoutManager.onMouseUp(LayoutManager.java:242)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.guihook.GuiContainerManager.mouseUp(GuiContainerManager.java:465)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.guihook.GuiContainerManager.overrideMouseUp(GuiContainerManager.java:458)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.client.gui.inventory.GuiContainer.func_146286_b(GuiContainer.java)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:306)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.client.gui.inventory.GuiContainer.func_146274_d(GuiContainer.java)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:268)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1640)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:973)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:3110)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.client.main.Main.main(SourceFile:148)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.launchwrapper.Launch.launch(Launch.java:165)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at net.minecraft.launchwrapper.Launch.main(Launch.java:29)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:243)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:278)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at org.multimc.EntryPoint.main(EntryPoint.java:34)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: Caused by: java.util.ConcurrentModificationException: java.util.ConcurrentModificationException: java.util.ConcurrentModificationException
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:562)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinTask.reportExecutionException(ForkJoinTask.java:604)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	... 35 more
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: Caused by: java.util.ConcurrentModificationException: java.util.ConcurrentModificationException
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:562)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:591)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:689)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:927)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at codechicken.nei.recipe.RecipeHandlerQuery.lambda$getHandlersWithRecipes$4(RecipeHandlerQuery.java:83)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1428)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: Caused by: java.util.ConcurrentModificationException
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1631)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:960)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:934)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
[20:27:14] [Client thread/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:763]: 	... 5 more
```
  
</details>

**Comment by Jerphf** [comment]
All recipes, I haven’t touched any configs and followed the migration/installation guide as close as possible. The only NEI setting I changed was the mode to test in creative.

**Comment by Jerphf** [comment]
After reading a post from 2013, I decided to try deleting the NEI save folder. I loaded in my survival world, opened my inventory, and tried searching for a recipe. It loaded a recipe with no textchat error. This was also the case with server utilities loaded. 
![server utilities 2 0 11 loaded](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/157259858/9135bd35-940b-4472-a1f3-d759337c1702)
![notextchaterror actually loads](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/157259858/1ab5bcb2-6f15-49ca-8de5-d921f6af90c1)

I'm guessing the file didn't migrate properly, but would work fine until I added server utilities. By deleting the migrated file, the game had to make a fresh one thus solving the issue. 

I honestly don't know why this worked, but it did for me.
